src/typing/type_inference_js.ml

```diff
@@ -930,8 +930,12 @@ and convert_qualification ?(lookup_mode=ForType) cx reason_prefix
 )
 
 and mk_rest cx = function
-  | ArrT(_, t, []) -> RestT t
+  | ArrT (_, t, []) -> RestT t
   | AnyT _ as t -> RestT t
+  | ArrT (r, t, _) ->
+      let msg = "Tuple type should not be used to annotate rest parameters" in
```

Comment:
This says what can't be done, but maybe you can specify that `Array<T>` is what can?